Obsidian.API/_Types/Traversal.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Obsidian.API;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Obsidian.API
+{
+    public class Traverse
+    {
+        private VectorF StartPoint { get; }
+        private VectorF EndPoint { get; }
+
+        /// <summary>
+        /// Initializes instance of <see cref="Traverse"/> with given start and end points
+        /// </summary>
+        /// <param name="startPoint">Start point of traversal</param>
+        /// <param name="endPoint">End point of traversal</param>
+
+        public Traverse(VectorF startPoint, VectorF endPoint)
+        {
+            StartPoint = startPoint;
+            EndPoint = endPoint;
+        }
+
+        private static float Frac0(float x)
+        {
+            return x - MathF.Floor(x);
+        }
+        private static float Frac1(float x)
+        {
+            return 1 - x + MathF.Floor(x);
+        }
+
+        /// <summary>
+        /// Runs voxel traversing
+        /// </summary>
+        /// <returns>IEnumerable of intersected positions</returns>
+        public IEnumerable<VectorF> Run()
+        {
+
+            List<VectorF> blocks = new List<VectorF>();
+
+            float tMaxX, tMaxY, tMaxZ, tDeltaX, tDeltaY, tDeltaZ;
+            VectorF voxel = new VectorF();
+
+            float x1 = StartPoint.X, y1 = StartPoint.Y, z1 = StartPoint.Z; // start point   
+            float x2 = EndPoint.X, y2 = EndPoint.Y, z2 = EndPoint.Z; // end point   
+
+
+            var dx = Math.Sign(x2 - x1); // x direction
+            if (dx != 0) tDeltaX = Math.Min(dx / (x2 - x1), 10000000.0f); else tDeltaX = 10000000.0f;
```

Comment:
`10000000.0f` seems to be a magic number. Could it be `float.MaxValue` or `float.PositiveInfinity`?

Reply:
Done